src/util/util.js

```diff
@@ -309,6 +309,33 @@ function getBaseUrl() {
   return (process.env?.HTTP_ONLY === 'true' ? 'http://' : 'https://') + process.env.DOMAIN;
 }
 
+/*
+ * Returns the root relative path, based on the referrer header. Since referrer header can be manipulated,
+ * we only return a non-null value if it is not to another domain
+ */
+const getSafeReferrer = (req /* Request */) => {
```

Comment:
Referer's cannot be trusted in general. So we only allow it as a referer if it belongs to the application domain. That allows these cases:
1. The referer is relative (eg /foobar or foobar) because the use of getBaseUrl() in URL.parse makes it relative to the app
2. The referer is absolute and the host after parsing the url matches the application domain or its www subdomain